### PR TITLE
feature: add pytest suite for auth, geolocation, and product listing services

### LINKED ISSUE #116: [TEST-2] Write Unit Tests
**As a** developer,  
**I want to** write unit tests for individual functions, services, and components,  
**so that** I can verify each part of the system works correctly in isolation.

### Acceptance Criteria

- Unit tests are created for core business logic and utility functions.
- Each test checks one clear behaviour or expected result.
- Tests include normal cases, edge cases, and invalid input cases.
- External dependencies such as databases, APIs, or services are mocked.
- All unit tests can be run automatically using the project test command.
- Failed tests clearly show which function or behaviour is incorrect.

### PR DESCRIPTION
**Pull request by Roshss2508**
Adds automated unit tests under tests/ using pytest, targeting:

- **auth** sign-in/sign-up/password logic, 
- **geolocation** normalization and distance helpers (with HTTP/API mocked),and 
- **product listing** serialization/search query building (with SQLAlchemy query chains mocked). 

Includes pytest.ini, shared Flask app context in conftest.py, and pytest in requirements.txt so python -m pytest tests -v runs the full suite locally and in CI.

image

Closes: #116

### CODE REVIEW

**Review by husthunterpy01 — COMMENTED** [review]
You should divide by folder for unit test and integration test, else it's hard to track

**Comment by Roshss2508** [comment]
Cool. I will do it.

File: tests/test_auth_service.py [hunk 1]
```diff
@@ -0,0 +1,193 @@
...
+from unittest.mock import MagicMock, patch
+
+import pytest
+from flask import Flask
+from werkzeug.security import generate_password_hash
+
+from app.service.authservice import AuthService
+
+
+@pytest.fixture
```
Comment (husthunterpy01) [inline tests/test_auth_service.py]: This looks like duplicate to conftest.py, please migrate this part to that file. A test file is only a test file

File: tests/test_auth_service.py [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+        assert err == "User does not exist"
+
+    def test_rejects_wrong_password(self):
+        existing = MagicMock()
+        existing.password = generate_password_hash("correct")
+        q = MagicMock()
+        q.first.return_value = existing
+        with patch("app.service.authservice.User.query") as uq:
+            uq.filter_by.return_value = q
+            user, err = AuthService.signin_user("someone@example.com", "wrong")
```
Comment (husthunterpy01) [inline tests/test_auth_service.py]: Can you update all the mock data into uwa email template, note that please fix it in the seed file as well

**Review by husthunterpy01 — CHANGES_REQUESTED** [review]

**Comment by Roshss2508** [comment]
## Summary
- Split automated tests into `tests/unit` (current suite) and `tests/integration` (placeholder for later).
- Moved duplicated Flask fixtures into `tests/conftest.py` per review — test files now contain tests only.
- Replaced generic/example emails in auth unit mocks with UWA-style addresses; updated `app/seed.py` user emails to match (`@student.uwa.edu.au` for students, `@uwa.edu.au` for admin accounts).
- Set `pytest.ini` `testpaths` to `tests/unit` so the default test command only runs unit tests until integration tests exist.

## Test plan
- [ ] `python -m pytest -v` (from repo root; picks up `tests/unit` via `pytest.ini`)

**Review by husthunterpy01 — APPROVED** [review]
Approve for merging
